src/components/modal/Modal.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { ReactNode, RefObject, useEffect, useState } from 'react';
+import ReactDOM from 'react-dom';
+import { IconAlertCircle, IconCross } from 'hds-react';
+import { PrimaryButton, SecondaryButton } from '../button/Button';
+import './Modal.scss';
+
+const ModalBackDrop = (): JSX.Element => <div className="modal-backdrop" />;
+const ModalWrapper = ({
+  id,
+  ariaLabelledBy,
+  children,
+}: {
+  id: string;
+  ariaLabelledBy: string;
+  children: ReactNode;
+}): JSX.Element => (
+  <div
+    role="alertdialog"
+    id={id}
+    className="modal-wrapper"
+    aria-labelledby={ariaLabelledBy}
+    aria-modal="true">
+    {children}
+  </div>
+);
+
+type UseModalProps = {
+  isOpen: boolean;
+  toggle: () => void;
+};
+
+export const useModal = (): UseModalProps => {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const toggle = (): void => setIsOpen(!isOpen);
+  return {
+    isOpen,
```

Comment:
Since these variables and functions are applied outside of this component, I wonder would it be better to name them more descriptive? isModalOpen and toggleModal? 🤔

Reply:
Will rename.